components/modal/index.js

```diff
@@ -0,0 +1,153 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import { noop } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { Component, createPortal } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { withModalContext } from './context';
+import ModalFrame from './frame';
+import ModalHeader from './header';
+import * as ariaHelper from './aria-helper';
+import './style.scss';
+
+// Used to count the number of open modals.
+let parentElement,
+	modalCount = 0;
+
+class Modal extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.node = document.createElement( 'div' );
+	}
+
+	static setAppElement( node ) {
+		ariaHelper.setAppElement( node );
+	}
+
+	static setParentElement( node ) {
+		if ( ! parentElement ) {
+			parentElement = node;
+		}
+	}
+
+	componentDidMount() {
+		modalCount++;
+
+		const {
+			modalContext,
+		} = this.props;
+
+		if ( ! this.parentElement ) {
+			setElements( modalContext.appElementId );
+		}
+
+		ariaHelper.hideApp();
```

Comment:
Doing this on mount can be dangerous because it assumes developers will mount the modal properly. For example, assuming there's a controlling app with a state property `showModal` if they do this:
`{ this.state.showModal && <Modal ...`

everything works fine because the modal is not mounted on page load and `aria-hidden` is not set.
But
if they do just this:
`<Modal ...`

the modal will be mounted (but not visible) on page load, and `aria-hidden`  will be set on the app element, with devastating consequences: the whole content of the page won't be available to assistive technologies.
I think having to require this correct usage is too fragile and prone to errors, we should try a safer way e.g.: setting aria-hidden only when isOpen is true.

Reply:
I fixed the logic behind this.